File: 242-valid-anagram/242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int>alph(26, 0);
        //cout << ('a' - 97);
        
        if(s.size() != t.size()) {return false;}
        for(int i = 0; i  < s.size(); i++) {
            alph[s[i]-97]++;
            alph[t[i]-97]--;
        }
        for(int i = 0 ; i < alph.size(); i++) {
            if(alph[i] != 0) {
                return false;
            }
        }
        
        
        return true;
    }
};

File: 242-valid-anagram/242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        let = {}
        
        for i in s:
            if i in let:
                let[i] += 1
            else:
                let[i] = 1
        
        for i in t:
            if i not in let:
                return False
            else:
                let[i] -= 1
        
        for i in let:
            if let[i] != 0:
                return False
            
        return True